fe/fe-core/src/main/java/org/apache/doris/nereids/rules/rewrite/EliminateGroupBy.java

```diff
@@ -0,0 +1,80 @@
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package org.apache.doris.nereids.rules.rewrite;
+
+import org.apache.doris.nereids.rules.Rule;
+import org.apache.doris.nereids.rules.RuleType;
+import org.apache.doris.nereids.trees.expressions.Alias;
+import org.apache.doris.nereids.trees.expressions.IsNull;
+import org.apache.doris.nereids.trees.expressions.NamedExpression;
+import org.apache.doris.nereids.trees.expressions.Slot;
+import org.apache.doris.nereids.trees.expressions.functions.agg.AggregateFunction;
+import org.apache.doris.nereids.trees.expressions.functions.agg.Count;
+import org.apache.doris.nereids.trees.expressions.functions.agg.Max;
+import org.apache.doris.nereids.trees.expressions.functions.agg.Min;
+import org.apache.doris.nereids.trees.expressions.functions.agg.Sum;
+import org.apache.doris.nereids.trees.expressions.functions.scalar.If;
+import org.apache.doris.nereids.trees.expressions.literal.Literal;
+import org.apache.doris.nereids.util.PlanUtils;
+
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+/**
+ * Eliminate GroupBy.
+ */
+public class EliminateGroupBy extends OneRewriteRuleFactory {
+
+    @Override
+    public Rule build() {
+        return logicalAggregate()
+                .when(agg -> agg.getGroupByExpressions().stream().allMatch(expr -> expr instanceof Slot))
+                .then(agg -> {
+                    Set<Slot> groupby = agg.getGroupByExpressions().stream().map(e -> (Slot) e)
+                            .collect(Collectors.toSet());
+                    boolean unique = agg.getLogicalProperties().getFunctionalDependencies().isUnique(groupby);
+                    if (!unique) {
+                        return null;
+                    }
+                    Set<AggregateFunction> aggregateFunctions = agg.getAggregateFunctions();
+                    if (!aggregateFunctions.stream().allMatch(
+                            f -> (f instanceof Sum || f instanceof Count || f instanceof Min || f instanceof Max)
+                                    && (f.arity() == 1 && f.child(0) instanceof Slot))) {
+                        return null;
+                    }
+
+                    List<NamedExpression> newOutput = agg.getOutputExpressions().stream().map(ne -> {
+                        if (ne instanceof Alias && ne.child(0) instanceof AggregateFunction) {
+                            AggregateFunction f = (AggregateFunction) ne.child(0);
+                            if (f instanceof Sum || f instanceof Min || f instanceof Max) {
+                                return new Alias(ne.getExprId(), f.child(0), ne.getName());
+                            } else if (f instanceof Count) {
+                                return (NamedExpression) ne.withChildren(
+                                        new If(new IsNull(f.child(0)), Literal.of(0), Literal.of(1)));
```

Comment:
It's wrong for empty children

Reply:
It is right.

such as  for empty children.

```sql
create table test (id int, age int);

select min(id) from test group by age;
select id from test;
```

They are same;